pkg/kstatus/polling/statusreaders/common.go

```diff
@@ -209,3 +209,15 @@ func errIdentifierToResourceStatus(err error, identifier object.ObjMetadata) (*e
 		Error:      err,
 	}, nil
 }
+
+func errWouldExceedContextDeadline(err error) bool {
+	for {
+		next := errors.Unwrap(err)
+		if next == nil {
+			break
+		}
+		err = next
+	}
+
+	return err != nil && err.Error() == "rate: Wait(n=1) would exceed context deadline"
```

Comment:
@karlkfi unwrapped recursively, could you have a look?

Reply:
i dont love it, but it'll work.